controllers/templatesync/template_sync.go

```diff
@@ -552,7 +675,28 @@ func (r *PolicyReconciler) Reconcile(ctx context.Context, request reconcile.Requ
 
 			successMsg := fmt.Sprintf("Policy template %s was updated successfully", tName)
 
-			err = r.handleSyncSuccess(ctx, instance, tIndex, tName, successMsg, res)
+			// Handle cluster scoped objects
+			if isClusterScoped {
+				addFinalizer = true
+
+				reqLogger.V(2).Info("Finalizer required for " + gvk.Kind)
+
+				// Ensure the CRD is labeled as policy-type=template so we can query custom resources for
+				// cleanup
+				err := applyCRDLabel(ctx, instance, tName, rsrc, dClient)
+				if err != nil {
+					tLogger.Error(err, fmt.Sprintf("Failed to apply %s label to CRD", utils.PolicyTypeLabel))
+				}
+
+				// The ConstraintTemplate does not generate status, so we need to generate an event for it
+				if isGkConstraintTemplate {
+					tLogger.Info("Emitting status event for " + gvk.Kind)
+					msg := fmt.Sprintf("%s %s was updated successfully", gvk.Kind, tName)
+					r.emitTemplateSuccess(instance, tIndex, tName, isClusterScoped, msg)
+				}
+			}
```

Comment:
Is this also at line 500?

Reply:
Ah, one is create, one is update. It's a bad code smell, but so is the fact that this Reconcile function is over 700 lines long...

Definitely something to address in the future.

Reply:
Definitely. There's a lot of replicated code--I tried to do my part to not make it too much worse by creating functions for repeated occurrences, but for sure could have gone a step further...